alfresco-search/src/test/java/org/alfresco/solr/tracker/DistributedCascadeTrackerTest.java

```diff
@@ -0,0 +1,207 @@
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.alfresco.solr.tracker;
+
+import org.alfresco.model.ContentModel;
+import org.alfresco.solr.AbstractAlfrescoDistributedTest;
+import org.alfresco.solr.client.Acl;
+import org.alfresco.solr.client.AclChangeSet;
+import org.alfresco.solr.client.AclReaders;
+import org.alfresco.solr.client.Node;
+import org.alfresco.solr.client.NodeMetaData;
+import org.alfresco.solr.client.StringPropertyValue;
+import org.alfresco.solr.client.Transaction;
+import org.alfresco.util.Pair;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.TermQuery;
+
+import org.apache.lucene.util.LuceneTestCase;
+import org.apache.solr.SolrTestCaseJ4;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Properties;
+
+
+import static org.alfresco.solr.AlfrescoSolrUtils.ancestors;
+import static org.alfresco.solr.AlfrescoSolrUtils.getAcl;
+import static org.alfresco.solr.AlfrescoSolrUtils.getAclChangeSet;
+import static org.alfresco.solr.AlfrescoSolrUtils.getAclReaders;
+import static org.alfresco.solr.AlfrescoSolrUtils.getNode;
+import static org.alfresco.solr.AlfrescoSolrUtils.getNodeMetaData;
+import static org.alfresco.solr.AlfrescoSolrUtils.getTransaction;
+import static org.alfresco.solr.AlfrescoSolrUtils.indexAclChangeSet;
+import static org.alfresco.solr.AlfrescoSolrUtils.list;
+import static org.carrot2.shaded.guava.common.collect.ImmutableList.of;
+
+/**
+ * @author Elia
+ */
+@SolrTestCaseJ4.SuppressSSL
+@LuceneTestCase.SuppressCodecs({"Appending","Lucene3x","Lucene40","Lucene41","Lucene42","Lucene43", "Lucene44", "Lucene45","Lucene46","Lucene47","Lucene48","Lucene49"})
+public class DistributedCascadeTrackerTest extends AbstractAlfrescoDistributedTest
+{
+    private Node parentFolder;
+    private NodeMetaData parentFolderMetadata;
+
+    private Node childShard0;
+    private NodeMetaData childShardMetadata0;
+
+    private Node childShard1;
+    private NodeMetaData childShardMetadata1;
+
+    private final String pathParent = "pathParent";
+    private final String pathChild0 = "pathChild0";
+    private final String pathChild1 = "pathChild2";
+
+    private final int timeout = 100000;
+
+    @BeforeClass
+    private static void initData() throws Throwable
+    {
+        initSolrServers(2, "DistributedCascadeTrackerTest", getShardMethod());
+    }
+
+    @AfterClass
+    private static void destroyData() throws Throwable
+    {
+        dismissSolrServers();
+    }
+
+
+    @Before
+    public void indexData() throws Exception
```

Comment:
I would put a comment which explains what precondition we are trying to achieve for this test

Reply:
Done